poppy/wfe.py

```diff
@@ -323,3 +324,32 @@ def get_opd(self, wave, units='meters'):
         if units == 'waves':
             opd /= wave.wavelength.to(u.meter).value
         return opd
+
+
+class StatsticalPSDWFE(WavefrontError):
+    """
+    Statistical PSF WFE class from power law for optical noise.
+
+    Input WFE in radians RMS.
+    """
+
+    @utils.quantity_input(wfe=u.rad)
+    def __init__(self, name='PSD WFE', index=3.0, wfe=0.25, **kwargs):
+
+        super(WavefrontError, self).__init__(name=name, **kwargs)
+        self.index = index
+        self.wfe_rad = wfe
+
+    def get_opd(self, wave, units='meters'):
+
+        y, x = self.get_coordinates(wave)
+        rho, theta = _wave_y_x_to_rho_theta(y, x, self.radius.to(u.meter).value)
+        psd = np.power(rho, -self.index)
+        psd = psd / np.sum(psd) * np.square(self.wfe_rad / wave.wavelength.to(u.meter).value)
+
+        rndm_phase = np.random.random((len(y), len(x)))
+        rndm_psd = np.fft.fftshift(np.fft.fft2(np.fft.fftshift(rndm_phase)))
+        scaled = np.sqrt(psd) * rndm_psd
+        phase_screen = np.real_if_close(np.fft.ifftshift(np.fft.ifft2(np.fft.ifftshift(scaled))))
```

Comment:
This needs a proper normalization.

Reply:
Since this part is somewhat cryptic and math-y, it might help to add citations of any relevant reference doc for the equations you used in implementing this, or else acknowledge and give credit to any other package you adapted this from, etc.

Reply:
Also, what is the benefit of calling `np.real_if_close` if you're just going to return `phase_screen.real` on the next line? Is that redundant?